numpy/tests/test_public_api.py

```diff
@@ -759,15 +759,19 @@ def _check___qualname__(obj) -> bool:
     actual_obj = functools.reduce(getattr, qualname.split("."), module)
     return (
         actual_obj is obj or
+        # `obj` may be a bound method of `actual_obj`:
         (
-            # for bound methods check qualname match
-            module_name.startswith("numpy.random") and
+            hasattr(actual_obj, "__get__") == hasattr(obj, "__self__") and
+            actual_obj.__module__ == obj.__module__ and
```

Comment:
I wonder if we also want add here `inspect.ismethod(obj) and`?

Reply:
For me it complains about a few `typing` an `ma` members:

```
{'actual': 'Union', 'expected': 'Union', 'module': <module 'numpy.typing' from '/home/mtsokol/numpy/build-install/usr/lib/python3.11/site-packages/numpy/typing/__init__.py'>}
{'actual': 'Union', 'expected': 'Union', 'module': <module 'numpy.typing' from '/home/mtsokol/numpy/build-install/usr/lib/python3.11/site-packages/numpy/typing/__init__.py'>}
{'actual': 'ndarray', 'expected': 'ndarray', 'module': <module 'numpy.typing' from '/home/mtsokol/numpy/build-install/usr/lib/python3.11/site-packages/numpy/typing/__init__.py'>}
{'actual': 'remainder', 'expected': 'remainder', 'module': <module 'numpy.ma' from '/home/mtsokol/numpy/build-install/usr/lib/python3.11/site-packages/numpy/ma/__init__.py'>}
{'actual': 'prod', 'expected': 'prod', 'module': <module 'numpy.ma' from '/home/mtsokol/numpy/build-install/usr/lib/python3.11/site-packages/numpy/ma/__init__.py'>}
{'actual': 'divide', 'expected': 'divide', 'module': <module 'numpy.ma' from '/home/mtsokol/numpy/build-install/usr/lib/python3.11/site-packages/numpy/ma/__init__.py'>}
```

Reply:
ooops, it should be `and` for the two `hasattr`.  It seems `typing.Union` lets its subtype inherit that name.

`Union` is weird, because it let's its subclasses inherit it's own name, I'll just add those explicitly for now.  CC @jorenham in case you are interested `typing.Union[...].__name__/__qualname__` are still `typing.Union`.

The `ma` errors there look like real errors though, their `__module__` is incorrect.

Reply:
Nevermind.  I think for the typing, the type aliases also fail this, so maybe it's just best to keep the full `typing` skip...

Reply:
So for the ma failures, the module may be something you care about.  But what the test dislikes is that the objects were not identical (although equivalent).
But, I don't see a reason in not just making them identical here (although opted for making it `"product"` for that one).